listen/property-changes.js

```diff
@@ -70,8 +68,9 @@ PropertyChanges.prototype.getOwnPropertyChangeDescriptor = function (key) {
         });
     }
     var objectPropertyChangeDescriptors = this.__propertyChangeListeners__;
-    if (!object_owns.call(objectPropertyChangeDescriptors, key)) {
+    if (!objectPropertyChangeDescriptors.hasOwnProperty(key)) {
```

Comment:
I don’t understand the potential correctness issue, unless objectPropertyChangeDescriptors would override hasOwnProperty? 


> On May 27, 2015, at 6:02 PM, Kris Kowal <notifications@github.com> wrote:
> 
> In listen/property-changes.js <https://github.com/montagejs/collections/pull/126#discussion_r31195693>:
> 
> > @@ -70,8 +68,9 @@ PropertyChanges.prototype.getOwnPropertyChangeDescriptor = function (key) {
> >          });
> >      }
> >      var objectPropertyChangeDescriptors = this.__propertyChangeListeners__;
> > -    if (!object_owns.call(objectPropertyChangeDescriptors, key)) {
> > +    if (!objectPropertyChangeDescriptors.hasOwnProperty(key)) {
> Is there definitive evidence that this sacrifice in correctness (consider key === 'hasOwnProperty') gives you a performance win? Asking for a friend.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/montagejs/collections/pull/126/files#r31195693>.
> 


 <http://montagestudio.com/>
montagestudio.com <http://montagestudio.com/>	 	
Benoit Marchant <mailto:benoit@montagestudio.com>
CEO & Founder
1290 Oakmead Parkway, Ste 111
Sunnyvale, CA 94085
(1) 408.621.4874 mobile <tel:+1-408-621-4874>
Profile <http://www.linkedin.com/in/benoitmarchant> |  @benoitmarchant <http://twitter.com/benoitmarchant>